src/private_pgn/tokenstream.js

```diff
@@ -255,12 +268,22 @@ TokenStream.prototype.consumeToken = function() {
 
 	// Otherwise, the string is badly formatted with respect to the PGN syntax
 	else {
-		throw new exception.InvalidPGN(this._text, this._pos, i18n.INVALID_PGN_TOKEN);
+		throw new exception.InvalidPGN(this._text, this._pos, this._lineCount, i18n.INVALID_PGN_TOKEN);
 	}
 
 	return true;
 };
 
+TokenStream.prototype.skipToNextGame = function() {
+	while(this._pos < this._text.length) {
+		if(testAtPos(this, this._matchEndOfGame)) {
```

Comment:
This won't give the expected behavior if something that looks like a end-of-game token is found in a header or in a game comment.

Reply:
I'll let you handle all PGN parsing issues. I think you are fixing them already while improving exception handling.